scenario_player/main.py

```diff
@@ -180,5 +182,121 @@ def reclaim_eth(obj, min_age):
+
+    files = set()
+
+    scenario_names = scenario_name or []
+
+    for scenario_name in (scenario_names or []):
+        # The logs are located at .raiden/scenario-player/scenarios/<scenario-name> - make sure the path exists.
+        scenario_log_dir = raiden_dir.joinpath('scenario-player', 'scenarios', scenario_name)
+        if not scenario_log_dir.exists():
+            print(f"No log directory found for scenario {scenario_name} at {scenario_log_dir}")
+            continue
+
+        # Add all folders that haven't been added yet.
+        for path in scenario_log_dir.iterdir():
+            if path.is_dir():
+                files.add(path)
+
+        files.union(pack_n_latest_logs_for_scenario_in_dir(scenario_name, scenario_log_dir, pack_n_latest))
+
+    # Now that we have all our files, create a tar archive at the requested location.
+    archive_fpath = target_dir.joinpath(f'Scenario_player_Logs-{"-",join(scenario_names)}-{pack_n_latest or "all"}-latest.tar.gz')
+    with tarfile.open(str(archive_fpath), mode='w:gz') as archive:
+        for file in files:
+            archive.add(str(file))
+
+    print(f"Created archive at {archive_fpath}")
+    print(f"- {archive_fpath}")
+
+    with tarfile.open(str(archive_fpath)) as f:
+        for name in f.getnames():
+            print(f"- - {name}")
+
+    if post_to_rocket:
+        post_to_rocket_chat(archive_fpath)
+
+
+def pack_n_latest_logs_for_scenario_in_dir(scenario_name, scenario_log_dir: Path, n) -> set:
+    # Add all scenario logs requested. Drop any iterations older than pack_n_latest,
+    # or add all if that variable is 0.
+    scenario_logs = [path for path in scenario_log_dir.iterdir() if (path.is_file() and str(path).startswith(scenario_name))]
+    scenario_logs = sorted(scenario_logs, key=lambda x: x.stat().st_mtime, reverse=True)
+
+    # specifying `pack_n_latest=0` will add all scenarios.
+    max_range = n or len(scenario_logs)
+
+    files = set()
+
+    for i in range(max_range):
+        try:
+            files.add(scenario_logs[i])
+        except IndexError:
+            # We ran out of scenario logs to add before reaching the requested number of n latest logs.
+            print(
+                f'Only packed {i} logs of requested latest {n} '
+                f'- no more logs found for {scenario_name}!',
+            )
+            break
+
+    return files
+
+
+def post_to_rocket_chat(fpath):
+    try:
+        user = os.environ['RC_USER']
+        pw = os.environ['RC_PW']
+        room_id = os.environ['RC_ROOM_ID']
+    except KeyError:
+        raise RuntimeError('Missing Rocket Char Env variables!')
+
+    resp = requests.post(
+        'https://chat.brainbot.com/api/v1/login',
+        data={'username': user, 'password': pw}
+    )
+
+    token = resp.json()['data']['authToken']
+    user_id = resp.json()['data']['userId']
+    headers = {
+        'X-Auth-Token': token,
+        'X-User-Id': user_id,
+    }
+
+    with fpath.open('rb') as f:
```

Comment:
IMO those files should rather be uploaded to something intended to store files (e.g. a DO spaces bucket) and then a link posted to RC.

One reason is that it will become pretty difficult to find older logs as time goes by and additionally in our current setup all attachments posted to RC are stored inside the mongodb, which makes backups annoying as they keep growing with attachements.

Reply:
Agreed. For this iteration, this will have to do though.  The notification / reporting services is the first of the things I will add after the nightlies run without throwing hissiy fits. They will then only send download URLs.